### DISCUSSION TITLE
Redis connection issue: tooz.coordination.ToozConnectionError: Timeout reading from socket

### DISCUSSION THREAD

**Discussion by SuganJoe** [StackStorm/st2#5717] [Q&A]
1. Is it possible to connect st2 to an external Redis server ? - If yes please confirm the below configuration is correct ?

2. Does St2 require to connect to Redis Cluster via sentinel only ?

We are trying to connect to AWS elasticache Redis server (TLS enabled)using its host name like below

```
st2:
  config: |
  [coordination]
  url = redis://<password>@master.server:6379
```
  
and get error below
`tooz.coordination.ToozConnectionError: Timeout reading from socket`

**Comment by amanda11** [comment]
I don't know if AWS ElasticCache Redis has any specialities. You can connect to redis with or without sentinel.
On your machine running STackStorm (or pods) if you use the redis-cli can you confirm that you can connect to redis - this will allow you to check the connectivity between the stackstorm node and the redis node. If you can connect via the cli with the same details as you want to use in the ST2 url then you can check its ok.

However you might need the fix that is in the unstable 3.8 stream, for the SSL timeout issue - although I didn't test against redis cluster, I suspect that part of the fix might also be required. If you are using StackStorm 3.7 then you wouldn't have that fix.
